docs/source/markdown/podman-image-scp.1.md

```diff
@@ -0,0 +1,84 @@
+% podman-image-scp(1)
+
+## NAME
+podman-image-scp - Securely copy an image from one host to another
+
+## SYNOPSIS
+**podman image scp** [*options*] *name*[:*tag*]
+
+## DESCRIPTION
+**podman image scp** saves an image to either **docker-archive**, **oci-archive**, **oci-dir** (directory with oci manifest type), or **docker-dir** (directory with v2s2 manifest type) on the local machine, and then copies the saved image to a remote host specified either by **--remote** globally, or by a connection name which has been added using **podman system connection add**. if neither **--remote** or a **HOSTNAME** has been specified, **image scp** will execute a local copy and save. default format is **docker-archive**. **podman image scp** writes to STDOUT by default and can be redirected to a file using the **output** flag. The **quiet** flag suppresses the output when set. **podman image scp** utilizes **podman save** and **podman load on the specified host or locally**.
+Note: `:` is used to specify the saved file location on the remote host. if not specified will default to the same path as specified for the **--output**
+
+**podman [GLOBAL OPTIONS]**
+
+**podman imaege scp [GLOBAL OPTIONS]**
+
+**podman image scp [OPTIONS] NAME[:TAG] HOSTNAME[:FILELOCATION]**
+
+## OPTIONS
+
+#### **--compress**
+
+Compress tarball image layers when pushing to a directory using the 'dir' transport. (default is same compression type, compressed or uncompressed, as source)
+Note: This flag can only be set when using the **dir** transport i.e --format=oci-dir or --format=docker-dir
+
+#### **--format**=*format*
```

Comment:
I have no idea how this would be used,  I can see specifying the transport in the source or destination arguments.  Don't see how a user would use the format option.

Reply:
I think the syntax should be
```
podman image scp CONNECTIONNAME:alpine .
podman image scp myimage CONNECTIONNAME:remoteImage
podman image scp root@localhost:foobar .
podman image scp foobar 10.20.3.1:docker-daemon:image:tag
podman image scp oci-archive:/tmp/image:tag roo@localhost:
```

Reply:
So should we support copying to and from the remote @rhatdan? Also, for simplicity and security I made it so that only connections specified by `system connection add` will be successful. I did this because `root@192.168....` would not be enough to ssh in most cases you need to specify the RSA key, adding bulk to the command.

Reply:
podman image scp CONNECTIONNAME:alpine . 

Would copy the alpine image into my own local container/storage.

As a user of this tool, I would expect it to be able to copy to remote and copy from remote.  As a matter of fact, I would expect it to be able to copy from remote to a different remote.

root@192.168...  

Would cause the underlying ssh to prompt for username/password or use the underlying security mechanisms to do the copy.  This is my assumption.

Some of these might not be able to be completed in the first version, but this is what I would want in the long run.

Reply:
@rhatdan, was able to add support for new ssh connection as well as to/from secure copying.

Reply:
> I have no idea how this would be used, I can see specifying the transport in the source or destination arguments. Don't see how a user would use the format option.

To put it another way, Podman should take care of this and use a format that works, without the user having to care about formats and feature sets. (Right now the `dir:` transport would probably be best for preserving the original, unlike `docker-archive:` and `oci-archive:` it does not force a conversion to a specific format. It’s also the only file-based format that can represent signatures.)

Especially consider if we add a streaming/RPC implementation later, to avoid copies of data already present in the destination, we don’t really want users to have to manually specify a format.

Reply:
After working on it again last week I totally agree @mtrmac. I included format at first because `podman save` takes the format flag and I was trying to properly use `save` and `load ` as is. Podman save and load in conjunction with one another only use oci-dir or docker-dir I think.

Reply:
> ```
> podman image scp CONNECTIONNAME:alpine .
> podman image scp myimage CONNECTIONNAME:remoteImage
> podman image scp root@localhost:foobar .
> podman image scp foobar 10.20.3.1:docker-daemon:image:tag
> podman image scp oci-archive:/tmp/image:tag roo@localhost:
> ```

Seems reasonable to me, apart from the `:` ambiguity (host`:`local-name vs. transport`:`…). The original man page uses `::` as a separator, and there’s good precedent for that (`rsync`). If we want a CLI without `--option`s in the simple case, which seems attractive, please consider `::` instead of `:`.

> Podman should always assume that the default is container/storage.

Pedantically speaking I’d expect this to use _exactly_ the lookup / write paths of `podman run` / `podman tag` (or some similar precedent within Podman), not any new semantics, nor a reimplementation of the existing lookup with broadly similar but not exactly the same outcomes.